core/services/relay/evm/capabilities/log_event_trigger_test.go

```diff
@@ -0,0 +1,85 @@
+package logevent_test
+
+import (
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services/servicetest"
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+	commonmocks "github.com/smartcontractkit/chainlink-common/pkg/types/core/mocks"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/triggers/logevent"
+	coretestutils "github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/capabilities/testutils"
+)
+
+// Test for Log Event Trigger Capability happy path for EVM
+func TestLogEventTriggerEVMHappyPath(t *testing.T) {
+	th := testutils.NewContractReaderTH(t)
+
+	logEventConfig := logevent.Config{
+		ChainID:        th.BackendTH.ChainID.String(),
+		Network:        "evm",
+		LookbackBlocks: 1000,
+		PollPeriod:     1000,
+	}
+
+	// Create a new contract reader to return from mock relayer
+	ctx := coretestutils.Context(t)
+
+	// Fetch latest head from simulated backend to return from mock relayer
+	height, err := th.BackendTH.EVMClient.LatestBlockHeight(ctx)
+	require.NoError(t, err)
+	block, err := th.BackendTH.EVMClient.BlockByNumber(ctx, height)
+	require.NoError(t, err)
+
+	// Mock relayer to return a New ContractReader instead of gRPC client of a ContractReader
+	relayer := commonmocks.NewRelayer(t)
+	relayer.On("NewContractReader", mock.Anything, th.LogEmitterContractReaderCfg).Return(th.LogEmitterContractReader, nil).Once()
+	relayer.On("LatestHead", mock.Anything).Return(commontypes.Head{
+		Height:    height.String(),
+		Hash:      block.Hash().Bytes(),
+		Timestamp: block.Time(),
+	}, nil).Once()
+
+	// Create Log Event Trigger Service and register trigger
+	logEventTriggerService, err := logevent.NewTriggerService(ctx,
+		th.BackendTH.Lggr,
+		relayer,
+		logEventConfig)
+	require.NoError(t, err)
+
+	// Start the service
+	servicetest.Run(t, logEventTriggerService)
+
+	log1Ch, err := logEventTriggerService.RegisterTrigger(ctx, th.LogEmitterRegRequest)
+	require.NoError(t, err)
+
+	expectedLogVal := int64(10)
+
+	// Send a blockchain transaction that emits logs
+	go func() {
+		_, err =
+			th.LogEmitterContract.EmitLog1(th.BackendTH.ContractsOwner, []*big.Int{big.NewInt(expectedLogVal)})
+		require.NoError(t, err)
```

Comment:
Two things:
1. Fatal/require APIs are only meant to be used from the "main" goroutine the test was spawned on, but you can `assert` from this goroutine, to report the failure without doing a fatal exit.
2. Technically this goroutine should be tracked so that we can block on it completing, because it must not live past the lifetime of the test.
```suggestion
	done := make(chan struct{})
	t.Cleanup(func() { <-done })
	go func() {
		defer close(done)
		_, err =
			th.LogEmitterContract.EmitLog1(th.BackendTH.ContractsOwner, []*big.Int{big.NewInt(expectedLogVal)})
		assert.NoError(t, err)
```

Reply:
Sure, fixed these